shopyo/modules/box__bizhelp/announce/view.py

```diff
@@ -61,4 +61,45 @@ def list():
     context.update({
         'announcements': announcements
         })
-    return mhelp.render('list.html', **context)
\ No newline at end of file
+    return mhelp.render('list.html', **context)
+
+
+@module_blueprint.route("/<announce_id>/delete/check", methods=["GET"])
+@login_required
+def delete_check(announce_id):
+    announcement = Announcement.query.get(announce_id)
+    announcement.delete()
+    return mhelp.redirect_url(mhelp.method('list'))
+
+
+@module_blueprint.route("/<announce_id>/edit", methods=["GET", "POST"])
+@login_required
+def edit(announce_id):
+    context = mhelp.context()
+    announcement = Announcement.query.get(announce_id)
+    
+    form = AnnounceForm(obj=announcement)
+
+    context.update({
+        'dir':dir,
+        'announcement': announcement,
+        'form': form
+        })
+    return mhelp.render('edit.html', **context)
```

Comment:
if `announce_id` is not valid, form displayed is empty which is okay but on submit it will redirect to `//edit/check` route which is invalid. Hence we should not allow `announce_id` get request for edit route. Also Why is POST allowed here?

Reply:
Will fix